apps/test/unit/templates/plugins/detailsTest.js

```diff
@@ -0,0 +1,237 @@
+      'contents, which are sometimes further block elements\n' +
+      ':::';
+    const expected =
+      '<details><summary>summary-content</summary><p>contents, which are sometimes further block elements</p></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('can have a variable number of opening colons', () => {
+    const markdown =
+      ':::::::: details [summary-content]\n' +
+      'contents, which are sometimes further block elements\n' +
+      ':::::::::::::';
+    const expected =
+      '<details><summary>summary-content</summary><p>contents, which are sometimes further block elements</p></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('can render markdown syntax in the summary', () => {
+    const markdown =
+      '::: details [**summary** _content_]\n' +
+      'contents, which are sometimes further block elements\n' +
+      ':::';
+    const expected =
+      '<details><summary><strong>summary</strong> <em>content</em></summary><p>contents, which are sometimes further block elements</p></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('can render markdown syntax in the body', () => {
+    const markdown =
+      '::: details [summary-content]\n' +
+      '\n' +
+      '# Contents\n' +
+      '- can\n' +
+      '- be\n' +
+      '- markdown\n' +
+      ':::';
+    const expected =
+      '<details><summary>summary-content</summary><h1>Contents</h1><ul>\n' +
+      '<li>can</li>\n' +
+      '<li>be</li>\n' +
+      '<li>markdown</li>\n' +
+      '</ul></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('ignores trailing colons', () => {
+    // Look how pretty this can be!
+    const markdown =
+      '::::::::::::: details [summary-content] :::::::::::::\n' +
+      'contents, which are sometimes further block elements\n' +
+      ':::::::::::::::::::::::::::::::::::::::::::::::::::::';
+    const expected =
+      '<details><summary>summary-content</summary><p>contents, which are sometimes further block elements</p></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('ignores excess whitespace', () => {
+    const markdown =
+      ':::      details       [summary-content]          \n' +
+      '\n' +
+      'contents, which are sometimes further block elements\n' +
+      '\n' +
+      ':::';
+    const expected =
+      '<details><summary>summary-content</summary><p>contents, which are sometimes further block elements</p></details>\n';
+
+    const rendered = parser.sourceToHtml(markdown);
+    assert.equal(rendered, expected);
+  });
+
+  it('can nest', () => {
+    const markdown =
+      ':::: details [outer]\n' +
+      '::: details [inner]\n' +
```

Comment:
Do nested details elements need to be proceeded by fewer colons than their parent(s)? If so, is that mentioned anywhere?

Reply:
Sort of. Blocks are closed by a line of colons longer than or equal to the opening, so in order to make sure the line which closes the inner block only closes it and not the outer block, the inner block needs to be closed by a line that's too short to close the outer.

From [the original proposal](https://talk.commonmark.org/t/generic-directives-plugins-syntax/444):

> Analogous to fenced code blocks, an arbitrary number of colons greater or equal three could be used as long as the closing line is longer than the opening line. That way, you can even nest blocks (think divs) by using successively fewer colons for each containing block.

Reply:
Ah I figured I missed this somewhere. Thanks!